### DISCUSSION TITLE
Is the email verification necessary when using oauth2 provider?

### DISCUSSION THREAD

**Discussion by PredragPeranovic** [fief-dev/fief#266] [General]
Hi, 

What is bugging me is that after authenticating using Google as an OAuth2 provider I need to verify my email.

So it looks like this:
1. Open the browser in incognito mode;
1.  Hit Login in my app;
1.  Click Signup using Google on the Fief login page;
1. Enter Google username and password on Google login page;
1. Confirm Google Two-factor auth on the phone;
1.  Confirm my email address for Fief to send me verification code
1.  Open the mail client to read Fief verification code 
1. Enter the verification code on the web page;
1. Finally I got logged in.

I think that if I add an OAuth2 provider I can trust data from it, so steps 6,7, and 8 are not needed. It will be nice to have the option to disable them.

All best, 
and great work so far.

**Comment by fief-bailiff** [comment]
Hail, @PredragPeranovic 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to star our repository ⭐️

![star-fief](https://user-images.githubusercontent.com/1144727/192523552-dc2c31fa-ae94-42fa-9ab4-7d0ba51bb7b0.png)

Farewell!

**Comment by frankie567** [comment]
Actually, it depends 🙃 

Some providers like Google indeed gives the information the email was verified or not. However, some others may not give this information (or don't verify emails at all).

For now, we don't have any specific logic given the provider, so the path is always the same no matter the provider. But this may something we improve in the future.

**Reply by PredragPeranovic** [reply]
Thanks for the feedback,

I didn't know about providers that don't validate email, because I have only used Google and internal Keycloak.
